cmake/CxxQt.cmake

```diff
@@ -25,6 +25,10 @@ endfunction()
 #
 # GEN_SOURCES an output variable that generated C++ files are listed into
 function(cxx_qt_generate_cpp GEN_SOURCES)
+    # FIXME: cxx's build.rs fails without this.
```

Comment:
Is this something we should report upstream to CXX ? or is it to do with how CMake is being used here?

Reply:
I'm unsure. I tested cloning the cxx repository and building that locally on macOS and could not reproduce the issue, so I think it might have something to do with invoking cargo from CMake, but I don't know why that would make a difference.

Reply:
>  fatal error: 'string.h' file not found

That was the error I was running into.

Reply:
I tried replacing this line with `set(CMAKE_CXX_STANDARD 17)` but it did not work. Curiously, when I manually specified `-D CMAKE_CXX_STANDARD=17` on the command line it worked. 🤔

Reply:
Let's find out with the cxx maintainers think: https://github.com/dtolnay/cxx/issues/1020

Reply:
Lets leave this upstream for now, I'll likely open an issue for each of the FIXME / TODOs from this branch when it lands.